src/js/fluxcontroller.js

```diff
@@ -272,13 +273,26 @@ define(function (require, exports, module) {
 
                 actionDependencies.set(action, dependencies);
 
-                var reads = action.reads || locks.ALL_LOCKS,
-                    writes = action.writes || locks.ALL_LOCKS;
+                var reads,
+                    writes;
+
+                if (actionObject) {
+                    reads = actionObject.reads || locks.ALL_LOCKS;
+                    writes = actionObject.writes || locks.ALL_LOCKS;
+                } else {
```

Comment:
If there is a good reason for this, it would be less redundant to do something like `var actionObject = action.action || {}, reads = actionObject.reads || locks.ALL_LOCKS`, etc.